libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -242,9 +259,21 @@ bool AP_Camera::update_location(const struct Location &loc)
         // be called without a new GPS fix
         return false;
     }
+
     if (get_distance(loc, _last_location) < _trigg_dist) {
         return false;
     }
-    _last_location = loc;
-    return true;
+
+    if (_max_roll > 0 && ((unsigned) ahrs.roll_sensor/100.0f > _max_roll)) {
```

Comment:
This still has the fabsf() problem. I believe the code below would fix it.

if (_max_roll > 0 && labs(ahrs.roll_sensor/100)> _max_roll) {